utils/filters/filterHelpers.ts

```diff
@@ -0,0 +1,55 @@
+import { Prisma, RoleType } from '@prisma/client';
+import { canBecomeValidYear, isValidYear } from '../date/validations';
+import { canBecomeInteger } from '../numbers/validations';
+
+export function makeCompanyNameFilters(searchWords: string[]) {
+  // Do not want to return names that contain the year to prevent zero results.
+  return searchWords.length === 0
+    ? undefined
+    : searchWords
+        .filter((word) => !canBecomeValidYear(word))
+        .map((word) => ({
+          name: {
+            contains: word,
+            mode: Prisma.QueryMode.insensitive,
+          },
+        }));
+}
+
+export function makeRoleTitleFilters(searchWords: string[]) {
+  // Do not want to return titles that contain the year.
+  // This prevents unrelated titles that only match the year but not the other search words.
+  // Filtering for year is done separately.
```

Comment:
What does this mean?

Reply:
If we find titles that match `2023`, suppose we search `cloud 2023`, we still get all other titles with `2023` like `software engineering intern 2023` which is too wide.

I will change the comment.